### PR TITLE
Gets all animals for matching_id (451)

### LINKED ISSUE #451: 360 View:  get_animals() only handles first SL id
As noted by [Karla on 10/20](https://codeforphilly.slack.com/archives/G018PE4L5M3/p1634779970002800),  the 360 page for the listed contact does not show all the contact's adoptions as they have multiple SL IDs.

get_animals() assumes only one SL id per person. We need to wrap the per-animal loop with an outer loop testing each SL id associated with the matching_id so we get all the animals for the matching_id.

### PR DESCRIPTION
**Pull request by c-simpson**
I added the outer loop in `get_animals()` to test all SL ids associated with the matching_id but then I hit an issue in Adoptions.js/getLatestPets() where it's accessing `animal.Events.Time`: 
- Animal #  22538082  appears not to have have any associated  events at SL (imported from old system?) so animal.Events is undefined

I added a check in `getLatestPets()` and if animals.Events or animal.Events.Time is not defined, I return a 1 so that one sorts last . 
I do a similar check in createRows() to return '' for Adoption Date if we don't have it.  

There may be better ways to achieve this - feel free to improve!

Closes #451

### CODE REVIEW

File: src/client/src/pages/DataView360/View/components/Adoptions.js [hunk 1]
```diff
@@ -37,7 +37,13 @@ class Adoptions extends Component {
 
     getLatestPets(animals) {
         const latestPets = _.sortBy(animals, animal => {
-            return animal.Events.Time
+            if (typeof animal.Events === 'undefined' || typeof animal.Events.Time  === 'undefined' ){
+                return 1
+            }
+            else{
+                return animal.Events.Time
+            }
```
Comment (dcstlouis) [inline src/client/src/pages/DataView360/View/components/Adoptions.js:45]: Just to tighten it up a bit, I'd suggest replacing lines 40-45 with:
`return (animal.Events && animal.Events.Time) ? animal.Events.Time : 1 `

File: src/client/src/pages/DataView360/View/components/Adoptions.js [hunk 2]
```diff
@@ -46,11 +52,21 @@ class Adoptions extends Component {
...
         const result = _.map(data, (row, index) => {
             const photo = row.Photos[0]
+            let eventDate;
+            if (typeof row.Events === 'undefined' || typeof row.Events.Time === 'undefined' ){
+                eventDate = ''
+            }
+            else{
+                eventDate = moment.unix(row.Events.Time).format("YYYY-MM-DD")
+            }
+
```
Comment (dcstlouis) [inline src/client/src/pages/DataView360/View/components/Adoptions.js:62]: Same thought here for `eventDate`: 

`const eventDate = (row.Events && row.Events.Time) ? moment.unix(row.Events.Time).format("YYYY-MM-DD") : ''`

**Review by dcstlouis — APPROVED** [review]
These changes make sense to me! A couple of style comments, but ultimately I think it's good to go!

File: src/client/src/pages/DataView360/View/components/Adoptions.js (hunk 1 above)
Comment (c-simpson) [reply src/client/src/pages/DataView360/View/components/Adoptions.js:45]: Way more elegant!

**Pull request merged**